agent/agent_test.go

```diff
@@ -126,6 +126,24 @@ func (s *AgentSuite) TestAgentEndTaskShouldExit() {
 	s.Error(err)
 }
 
+func (s *AgentSuite) TestNextTaskConflict() {
+	s.mockCommunicator.NextTaskShouldConflict = true
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	errChan := make(chan error)
+	go func() {
+		errChan <- s.a.loop(ctx)
+	}()
+	time.Sleep(time.Millisecond)
+	select {
+	case err := <-errChan:
+		s.NoError(err)
```

Comment:
check that it's the right error.

Reply:
The way this works is that there should _not_ be an error. Since there is a task conflict, the loop should loop forever. For other errors, the loop returns, so if we got an error on the error channel, it means something has gone wrong. Any error should be a test failure.